src/node_dotenv.cc

```diff
@@ -42,19 +70,37 @@ std::vector<Dotenv::env_file_data> Dotenv::GetDataFromArgs(
       auto flag = matched_arg->substr(0, equal_char_index);
       auto file_path = matched_arg->substr(equal_char_index + 1);
 
+      auto sections = get_sections(file_path);
+
+      auto hash_char_index = file_path.find('#');
+      if (hash_char_index != std::string::npos) {
+        file_path = file_path.substr(0, hash_char_index);
+      }
+
       struct env_file_data env_file_data = {
-          file_path, flag.starts_with(optional_env_file_flag)};
+          file_path, flag.starts_with(optional_env_file_flag), sections};
       env_files.push_back(env_file_data);
     } else {
       // `--env-file path`
-      auto file_path = std::next(matched_arg);
+      auto file_path_ptr = std::next(matched_arg);
 
-      if (file_path == args.end()) {
+      if (file_path_ptr == args.end()) {
         return env_files;
       }
 
+      std::string file_path = file_path_ptr->c_str();
```

Comment:
This is making unnecessary copy. Technically you can construct a std::string_view and avoid a copy.